std-bits/table/src/main/java/org/enso/table/operations/AddRunning.java

```diff
@@ -41,38 +43,92 @@ private static RunningStatistic<Double> createRunningStatistic(
         return new RunningMeanStatistic(sourceColumn, problemAggregator);
       }
       case Minimum -> {
+        if (sourceColumn.getStorage().getType() == IntegerType.INT_64) {
+          return new RunningMinLongStatistic(sourceColumn, problemAggregator);
+        }
         return new RunningMinStatistic(sourceColumn, problemAggregator);
       }
       case Maximum -> {
+        if (sourceColumn.getStorage().getType() == IntegerType.INT_64) {
+          return new RunningMaxLongStatistic(sourceColumn, problemAggregator);
+        }
         return new RunningMaxStatistic(sourceColumn, problemAggregator);
       }
       default -> throw new IllegalArgumentException("Unsupported statistic: " + statistic);
     }
   }
 
-  private abstract static class RunningStatisticBase implements RunningStatistic<Double> {
+  private interface TypeHandler<T> {
```

Comment:
Note that usage of generics here will cause boxing of `double`/`long` values to their boxed `Double`/`Long` counterparts which may impact performance.

I cannot tell how big the impact here will be, but it may be better to have separate implementations - that's what we did for most operations.

Reply:
When you say separate implementations do you mean copy/paste the generic code for each type?

Reply:
Sadly, yes

Reply:
I wonder if we can do something clever using the storage but I think this is ok for this pass.